includes/widgets/class-amp-widget-archives.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Class AMP_Widget_Archives
+ *
+ * @since 0.7.0
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Widget_Archives
+ *
+ * @since 0.7.0
+ * @package AMP
+ */
+class AMP_Widget_Archives extends WP_Widget_Archives {
+
+	/**
+	 * Echoes the markup of the widget.
+	 *
+	 * Mainly copied from WP_Widget_Archives::widget()
+	 * Changes include:
+	 * An id for the <form>.
+	 * More escaping.
+	 * The dropdown is now filtered with 'wp_dropdown_cats.'
+	 * This enables adding an 'on' attribute, with the id of the form.
+	 * So changing the dropdown value will redirect to the category page, with valid AMP.
+	 *
+	 * @since 0.7.0
+	 *
+	 * @param array $args Widget display data.
+	 * @param array $instance Data for widget.
+	 * @return void.
+	 */
+	public function widget( $args, $instance ) {
+		if ( ! is_amp_endpoint() ) {
+			parent::widget( $args, $instance );
```

Comment:
Excellent move to defer the `is_amp_endpoint()` check to when the `widget` is actually called.